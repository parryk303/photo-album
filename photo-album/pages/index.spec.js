/**
 * @jest-environment jsdom
 */
 const { shallow, mount, render, configure } = require('enzyme')
 import renderer from 'react-test-renderer'

 import Index from './index'

 const sample = [
   {
     id: 1,
     title: "quidem molestiae enim",
     userId: 1
   },
   {
     id: 2,
     title: "sunt qui excepturi placeat culpa",
     userId: 1
   }
 ];

 describe('Index', () => {
   it('renders with no errors', () => {
     const data = sample;
     const wrapper = mount(<Index albums={data} results={data} />);
     expect(wrapper.props().albums[0]).toEqual(data[0]);
     expect(wrapper.props().albums[1]).toEqual(data[1]);
     expect(wrapper.props().results[0]).toEqual(data[0]);
     expect(wrapper.props().results[1]).toEqual(data[1]);
   });
 });

 describe('Index Snapshot', () => {
   it('renders with no errors', () => {
     const data = sample;
     const component = renderer.create(<Index albums={data} results={data} />);
     const tree = component.toJSON();
     expect(tree).toMatchSnapshot();
   });
 })
