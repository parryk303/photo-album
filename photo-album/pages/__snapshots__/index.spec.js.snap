// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Snapshot renders with no errors 1`] = `
<div
  className="container"
>
  <div
    className="ui ten stackable cards"
  >
    <div
      className="card"
      id="album"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <h2
        className="id"
      >
        1
         
        <i
          className="folder outline icon"
        />
         
      </h2>
    </div>
    <div
      className="card"
      id="album"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <h2
        className="id"
      >
        2
         
        <i
          className="folder outline icon"
        />
         
      </h2>
    </div>
  </div>
  <footer
    className="footer"
  >
    <a
      href="https://vercel.com?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
      rel="noopener noreferrer"
      target="_blank"
    >
      Powered by
       
      <span
        className="logo"
      >
        <span
          style={
            Object {
              "background": "none",
              "border": 0,
              "boxSizing": "border-box",
              "display": "inline-block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "overflow": "hidden",
              "padding": 0,
              "position": "relative",
              "width": "initial",
            }
          }
        >
          <span
            style={
              Object {
                "background": "none",
                "border": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "padding": 0,
                "width": "initial",
              }
            }
          >
            <img
              alt=""
              aria-hidden={true}
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNzIiIGhlaWdodD0iMTYiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "display": "block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "padding": 0,
                  "width": "initial",
                }
              }
            />
          </span>
          <img
            alt="Vercel Logo"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
          <noscript>
            <img
              alt="Vercel Logo"
              data-nimg="intrinsic"
              decoding="async"
              loading="lazy"
              src="/_next/image?url=%2Fvercel.svg&w=256&q=75"
              srcSet="/_next/image?url=%2Fvercel.svg&w=96&q=75 1x, /_next/image?url=%2Fvercel.svg&w=256&q=75 2x"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </noscript>
        </span>
      </span>
    </a>
  </footer>
</div>
`;
